core/tools/mavlink_router/bootstrap.py

```diff
@@ -35,6 +35,10 @@ def run_apt_install(package: str) -> None:
 
 
 def run_apt_uninstall(package: str) -> None:
+    if "NOUNINSTALL" in os.environ:
+        print("Skipping uninstall step.")
+        return
+
```

Comment:
I think it would make more sense for this skip statement to be on the call to the function, on line 79. It's more explicit there. Here we call uninstall and it does not uninstall, which is somewhat weird. 

What do you think?

Reply:
This is a global rule for the uninstall step, if the function is called more than once the same if would not to exist.

Reply:
Didn't get it. Why would someone need to call uninstall twice?

Reply:
To encapsulate point specific uninstalls for better organization, like: build time dependencies and runtime dependencies, as we have for GST in companion-core.

Reply:
Yes. I'm not suggesting to remove it, just to move it to the call of the function, instead of the function itself, like that:

```python
if "NOUNINSTALL" not in os.environ:
    run_apt_uninstall("autoconf g++ git libtool make pkg-config python3-future")
else:
    print("Skipping uninstall step.")
```

Reply:
Sorry if I was not clear, but as I said in my previous comments, multiple calls for the uninstall function may happen in the future and the idea behind NOUNINSTALL environment variable is to be a global behavior change and not scope specific, that's why it's inside the function and not outside as a scope specific behavior control.
If that's not clear yet, imagine the following scenario, that as I said previous, is based under the docker-base with GST installation.
```
BUILD_LIBS = [A, B, C]
RUNTIME_LIBS = [D, E, F]

// code
install(BUILD_LIBS)
//code for build
uninstall(BUILD_LIBS)

// more code
install(RUNTIME_LIBS)
//code for runtime
uninstall(RUNTIME_LIBS)
```

With the current patch this code does not change since we are controlling the *global* behavior and not scope specific behavior. Otherwise we'll need a if check for each uninstall.

Reply:
Totally get it. Just saying it sounds weird to me calling a function named "uninstall_something" and this something is not uninstalled.

But hey, as I said, its up to you. That's totally a minor.